src/main/resources/META-INF/plugin.xml

```diff
@@ -37,30 +40,37 @@
     <!-- Provide Metadata for Dummy Schemas -->
-    <metaDataContributor implementation="org.kohsuke.stapler.idea.descriptor.JellyTaglibMetadataContributor" />
+    <metaDataContributor implementation="org.kohsuke.stapler.idea.descriptor.JellyTaglibMetadataContributor"/>
     <!-- ? -->
-    <annotator language="XML" implementationClass="org.kohsuke.stapler.idea.JellyAnnotator" />
+    <annotator language="Jelly" implementationClass="org.kohsuke.stapler.idea.JellyAnnotator"/>
     <!-- Show Documentation for Jelly Tag Libraries
        https://plugins.jetbrains.com/docs/intellij/documentation.html#implementation
     -->
-    <lang.documentationProvider language="XML" implementationClass="org.kohsuke.stapler.idea.JellyDocumentationProvider"/>
+    <lang.documentationProvider language="Jelly" implementationClass="org.kohsuke.stapler.idea.JellyDocumentationProvider"/>
     <!-- Tells IntelliJ that script and style tags inside *.jelly files are JavaScript and CSS correspondingly -->
-    <multiHostInjector implementation="org.kohsuke.stapler.idea.JellyLanguageInjector" />
+    <multiHostInjector implementation="org.kohsuke.stapler.idea.JellyLanguageInjector"/>
     <!-- Evaluates JEXL and reports any errors -->
-    <localInspection language="XML" enabledByDefault="true" level="ERROR"
+    <localInspection language="Jelly" enabledByDefault="true" level="ERROR"
                      displayName="Checks syntax of JEXL expressions"
                      groupName="Stapler"
-                     implementationClass="org.kohsuke.stapler.idea.JexlInspection" />
+                     implementationClass="org.kohsuke.stapler.idea.JexlInspection"/>
     <!-- FooBar. No intention described in the javadocs -->
-<!--
-    <localInspection language="XML" enabledByDefault="true" level="ERROR"
-                     shortName="I18nInspection" displayName="Test something"
-                     groupName="Stapler"
-                     implementationClass="org.kohsuke.stapler.idea.I18nInspection" />
--->
-   </extensions>
+    <!--
+        <localInspection language="Jelly" enabledByDefault="true" level="ERROR"
+                         shortName="I18nInspection" displayName="Test something"
+                         groupName="Stapler"
+                         implementationClass="org.kohsuke.stapler.idea.I18nInspection" />
+    -->
+    <!-- jelly file definition parser -->
+    <lang.parserDefinition language="Jelly"
+                           implementationClass="org.kohsuke.stapler.idea.language.JellyParserDefinition"/>
+    <!-- Additional link to jelly file from a java file -->
+    <lang.structureViewExtension implementation="org.kohsuke.stapler.idea.extension.JavaStructureViewExtension"/>
+    <!-- Additional link to java file from a jelly file -->
+    <lang.structureViewExtension implementation="org.kohsuke.stapler.idea.extension.JellyStructureViewExtension"/>
```

Comment:
I've been on the fence about this for a while now. I am just unsure what kind of XML support features are lost versus what is gained by this. 

Given structure view extension is "attached" to a PSI class, you kinda needed this. That said, you could have probably just used the `XmlDocument.class` and did the `JellyFileTypeSchema.isJelly(parent.getContainingFile())` in the `getChildren` method.

Reply:
Actually the `JellyLanguage` extends the `XMLLanguage`, therefore all XML features are included in Jelly language.
With this, checking the file instance against `JellyFile` is enough.

More, this method `JellyFileTypeSchema.isJelly(parent.getContainingFile())` simply checks the extension and we redeclare `jelly` and `jellytag`, but we shouldn't anymore as it's already linked/declared [here](https://github.com/jgreffe/idea-stapler-plugin/blob/feat/102_structure-view/src/main/resources/META-INF/plugin.xml#L28).
i.e : if we add a new extension for jelly, like `jellyanothertag`, we would need to update `plugin.xml`, and also `JellyFileTypeSchema`, which sounds like a duplicate/overhead to me (?)

Reply:
I see what you mean. All the File Type, Language and PSI classes extend corresponding XML ones, so theoretically all the extensions that look for those types should still work with these derived types. I suspected that this type of change is necessary to get proper JEXL integration, but I am not sure, and JEXL support is just too much work. 

I got burned by doing incremental changes to the file types before. You can see it if you read through https://github.com/jenkinsci/idea-stapler-plugin/releases

There is also a question of what happens if the end-user made a manual association between `jelly` extension and XML file type. From my previous experience, manual assosciation takes precedence, so the introduction of a new file type might break some old setups.

I'll need to ponder and test some more.

> which sounds like a duplicate/overhead to me (?)

It was duplication even before your change. It looked unavoidable. I used JavaFX plugin for inspiration when I implemented the file type checking in that way. With the introduction of the Jelly File type, those public static predicates should no longer be needed.